File: xssfinder-java-executor/src/main/java/org/xssfinder/runner/CustomSubmitTraversalStrategy.java
package org.xssfinder.runner;

import org.xssfinder.CustomSubmitter;
import org.xssfinder.remote.TraversalMode;

import java.lang.reflect.Method;
import java.util.HashMap;
import java.util.Map;

public class CustomSubmitTraversalStrategy implements TraversalStrategy {
    private final CustomSubmitterInstantiator submitterInstantiator;
    private final LabelledXssGeneratorFactory labelledXssGeneratorFactory;

    public CustomSubmitTraversalStrategy(CustomSubmitterInstantiator submitterInstantiator, LabelledXssGeneratorFactory labelledXssGeneratorFactory) {
        this.submitterInstantiator = submitterInstantiator;
        this.labelledXssGeneratorFactory = labelledXssGeneratorFactory;
    }

    @Override
    public boolean canSatisfyMethod(Method method, TraversalMode traversalMode) {
        if (traversalMode != TraversalMode.SUBMIT) {
            return false;
        }
        CustomSubmitter customSubmitter = submitterInstantiator.instantiate(method);
        return customSubmitter != null;
    }

    @Override
    public TraversalResult traverse(Object page, Method method) {
        CustomSubmitter customSubmitter = submitterInstantiator.instantiate(method);
        Map<String, String> inputIdsToAttackIds = new HashMap<String, String>();
        LabelledXssGeneratorImpl generator = labelledXssGeneratorFactory.createLabelledXssGenerator();
        Object newPage = customSubmitter.submit(page, generator);
        inputIdsToAttackIds.putAll(generator.getLabelsToAttackIds());
        return new TraversalResult(newPage, inputIdsToAttackIds);
    }
}

File: xssfinder-php-executor/src/XssFinder/Runner/CustomSubmitTraversalStrategy.php
<?php

namespace XssFinder\Runner;

use ReflectionMethod;
use XssFinder\TraversalMode;
use XssFinder\TUntraversableException;

class CustomSubmitTraversalStrategy implements TraversalStrategy
{
    /** @var CustomSubmitterInstantiator */
    private $_customSubmitterInstantiator;

    public function __construct(CustomSubmitterInstantiator $customSubmitterInstantiator)
    {
        $this->_customSubmitterInstantiator = $customSubmitterInstantiator;
    }

    /**
     * @param ReflectionMethod $method
     * @param int $traversalMode
     * @return boolean
     */
    function canSatisfyMethod(ReflectionMethod $method, $traversalMode)
    {
        return $traversalMode === TraversalMode::SUBMIT && $this->_customSubmitterInstantiator->hasCustomSubmitter($method);
    }

    /**
     * @param mixed $page
     * @param ReflectionMethod $method
     * @return TraversalResult
     *
     * @throws TUntraversableException
     */
    function traverse($page, ReflectionMethod $method)
    {
        $submitter = $this->_customSubmitterInstantiator->instantiate($method);
        $newPage = $submitter->submit($page, null); //TODO: Pass LabelledXssGenerator
        $traversalResult = new TraversalResult($newPage, array()); //TODO: Return LXG's recorded results
        return $traversalResult;
    }
}